tasks/post_copy_request_to_queue/post_copy_request_to_queue.py

```diff
@@ -0,0 +1,228 @@
+"""
+Name: post_copy_request_to_queue.py
+Description:  lambda function that queries the db for file metadata, updates the status
+of recovered file to staged, and sends the staged file info to staged_recovery queue for further processing.
+
+"""
+from enum import Enum
+import json
+import os
+import boto3
+from typing import Dict, Any, List, Optional
+from datetime import datetime, timezone
+import time
+import random
+from shared_recovery import (
+    OrcaStatus,
+    RequestMethod,
+    post_status_for_file_to_queue,
+    post_entry_to_queue,
+)
+import requests_db
+import database
+from cumulus_logger import CumulusLogger
+import logging
+
+# instantiate CumulusLogger
+LOGGER = CumulusLogger()
+
+
+def task(
+    records: Dict[str, Any],
+    db_queue_url: str,
+    recovery_queue_url: str,
+    max_retries: int,
+    retry_sleep_secs: int,
+    retry_backoff: int,
+) -> None:
+    """
+    Task called by the handler to perform the work.
+
+    This task queries all entries from orca_recoverfile table
+    that match the given filename and whose status_id is 'PENDING'.
+    The result is then sent to the staged-recovery-queue SQS and status-update-queue SQS.
+
+    Args:
+        records: A dictionary passed through from the handler.
+        db_queue_url: The SQS URL of status_update_queue
+        recovery_queue_url: The SQS URL of staged_recovery_queue
+        max_retries: Number of times the code will retry in case of failure.
+        retry_sleep_secs: Number of seconds to wait between recovery failure retries.
+        retry_backoff: The multiplier by which the retry interval increases during each attempt.
+    Returns:
+        None
+    Raises:
+        Exception: If unable to retrieve key_path or db parameters, convert db result to json,
+        or post to queue.
+
+    """
+    # grab the key_path and bucketname from event
+    # We only expect one record.
```

Comment:
Is there a value in the aws_s3_bucket_notification that gives us only one value in a list?

Reply:
we will only have one item according to our design. I have added that as a comment above. Also talked with Brad regarding this

Reply:
But what in our terraform enforces that limit? See https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_event_source_mapping for the batch_size property. Apparently it defaults to 10.

Reply:
changed to records[0]["s3"]["object"]["key"]

Reply:
But if the batch size defaults to 10, then this will only process the first item. Please explain to me how you know that only one item at a time will enter this lambda.

Reply:
we should change the [batch_size](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_event_source_mapping#batch_size) parameter for resource "aws_lambda_event_source_mapping to 1 instead of 10 in terraform config. Then only one item will enter lambda at a time.

The batch_size is only applicable when the message goes from SQS-> lambda. In post_copy_request_to_queue, that is not the case. We should change this in terraform config of copy_files_to_archive lambda